Makefile

```diff
@@ -2,7 +2,15 @@
 
 .DEFAULT_GOAL := all
 
-include .mage/tools.mk
+mage_dir := .mage
+magefile := $(mage_dir)/tools/mgmake/magefile
+
+$(magefile): $(mage_dir)/go.mod $(shell find $(mage_dir)/.. -type f -name '*.go')
+	@cd $(mage_dir) && go run go.einride.tech/mage-tools gen
+
+.PHONY: mage-clean
+mage-clean:
+	@git clean -fdx $(mage_dir)
 
 .PHONY: all
 all: $(mage)
```

Comment:
the `$(mage)` here should also get updated?

Reply:
Fixed